main.tf

```diff
@@ -42,32 +42,33 @@ module "cluster" {
 }
 
 module "service" {
-  for_each = { for k, v in var.service_map : k => v if v.create }
+  for_each = { for k, v in var.service_map : k => v if lookup(v, "create", true) }
+
+  source = "./modules/service"
 
-  source                = "./modules/service"
   name                  = format("%s-%s", var.name, replace(each.key, "_", "-"))
   cluster_id            = module.cluster.ecs_cluster_id
-  container_definitions = each.value.service_container_definitions
+  container_definitions = lookup(each.value, "service_container_definitions", null)
   launch_type           = var.launch_type
-  task_cpu              = each.value.service_task_cpu
-  task_memory           = each.value.service_task_memory
-  desired_count         = each.value.service_desired_count
-  execution_role_arn    = var.service_task_execution_role_arn
-  task_role_arn         = var.service_task_role_arn
+  task_cpu              = lookup(each.value, "service_task_cpu", null)
+  task_memory           = lookup(each.value, "service_task_memory", null)
+  desired_count         = lookup(each.value, "service_desired_count", null)
+  execution_role_arn    = lookup(each.value, "execution_role_arn", var.service_task_execution_role_arn)
+  task_role_arn         = lookup(each.value, "task_role_arn", var.service_task_role_arn)

```

Comment:
Actually, these should be mandatory right?

Reply:
yes. need to check which are, container_definition for sure mandatory.

https://github.com/SPHTech-Platform/terraform-aws-ecs/pull/3#discussion_r1007557654

Reply:
task_cpu and task_memory are required for ecs fargate.

Reply:
Changed some of them. Good practice to always define task cpu and memory anyway